### PR TITLE
Add some JSDoc to bin/pa11y.js

### PR DESCRIPTION
**Pull request by josebolos**
There's a bunch of functions on this file, so now at least it will be easier to understand what each one of them is doing, as there are no unit tests for them.

### CODE REVIEW

File: bin/pa11y.js [hunk 1]
```diff
@@ -124,9 +132,14 @@ async function runProgram() {
 	}
 }
 
+/**
+ * Process options based on precedence rules.
+ * CLI options take precedence over config options (which take precedence over defaults)
+ * @returns {Object} An object containing all the options that have been specified
```
Comment (joeyciechanowicz) [inline bin/pa11y.js]: You could do a `@typedef` and define the full config object....if you _really_ cared... 🤔

File: bin/pa11y.js [hunk 2]
```diff
@@ -203,6 +241,14 @@ function requireFirst(stack, defaultReturn) {
 	}
 }
 
+/**
+ * Calculates if there are any errors to be reported (i.e. above the threshold)
+ * This is used for example to return the correct status code on the CLI
+ * @param {String} level - The level of reporting required (warning, notice, etc.)
+ * @param {Object} issues - The issues found by the test
```
Comment (joeyciechanowicz) [inline bin/pa11y.js]: The parameter is named `results`
```suggestion
 * @param {Object} results - The results found by the test
```

File: bin/pa11y.js [hunk 3]
```diff
@@ -224,10 +270,21 @@ function isWarningOrError(result) {
 	return (result.type === 'warning' || result.type === 'error');
 }
 
+/**
+ * Create an array from a semicolon-separated string of options
+ * @param {String} val - The string of user-specified values
+ * @param {Array} array - The previous value, if any
```
Comment (joeyciechanowicz) [inline bin/pa11y.js]: Needs to be plural
```suggestion
 * @param {Array} array - The previous value(s), if any
```

File: bin/pa11y.js (hunk 1 above)
Comment (josebolos) [reply bin/pa11y.js]: I don't! :trollface:

**Pull request merged**